### ISSUE TITLE
[Feature Request] Loop_function argument for seq2seq models

### ISSUE THREAD

**Issue by Conchylicultor** [tensorflow/tensorflow#5391]
The RNN decoder has a [parameters](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L117) `loop_function` which allows to do some pre-processing on the output of the RNN before connecting the output on the next input.

That would be great to have this parameters for `embedding_rnn_seq2seq` and `embedding_rnn_decoder`. For now the only control we can have is no loop function at all when `feed_previous=False` or an argmax on the output called internally [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L274) .

The loop function parameters should be really useful to sample from the softmax distribution, instead of only taking the most likely output (as it is the case now with `feed_previous`) or to implement [scheduled sampling](https://arxiv.org/abs/1506.03099). 

For now the only solution is to re-create custom versions of `embedding_rnn_decoder` `embedding_rnn_seq2seq` just to change this parameters.

**Comment by aselle** [comment]
@ebrevdo, could you consider this suggestion?

**Comment by ebrevdo** [comment]
We are working on a new seq2seq api in tf.contrib.seq2seq which should be flexible enough for you to do this.

**Comment by Conchylicultor** [comment]
Great. Thanks!

**Comment by Conchylicultor** [comment]
I close the issue as this can now be done with the new seq2seq API and the `decoder_fn` parameter.
